### PR TITLE
feat: add molgenis auth as dep

### PR DESCRIPTION
**Pull request by sidohaakma**

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -6,25 +6,30 @@ Authors@R:
     c(person(given = "Sido",
              family = "Haakma",
              role = c("aut", "cre"),
-             email = "s.haakma@rug.nl"),
+             email = "s.haakma@rug.nl"
+             comment = c(ORCID = "0000-0003-1278-9144")),
```
Comment (fdlk) [inline DESCRIPTION:10]: whoo!

File: vignettes/DSArmadillo.Rmd [hunk 2]
```diff
@@ -1,79 +1,116 @@
...
 There is a default workflow in DataSHIELD to do analysis. There are several steps that you need to take.
 
-- Login to the Armadillo
+- Get token from the central authentication server
+- Log in to the Armadillo
 - Assign data
 - Transform data (if necessary)
 - Performing analysis
-- Workspaces
+- Work spaces
```
Comment (fdlk) [inline vignettes/DSArmadillo.Rmd:71]: typo? I do think it's one word

File: tests/testthat/test-ArmadilloOAuth.R [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  credentials <- mock(list(id_token = "abcd-abcd"))
+
+  with_mock(
+    "httr::GET" = get,
+    "httr::content" = content,
+    "MolgenisAuth::discover" = discover,
+    "MolgenisAuth::device_flow_auth" = credentials,
+    token <- datashield.get.token("https://armadillo.dev.molgenis.org")
+  )
+  expect_equal(token, "abcd-abcd")
```
Comment (fdlk) [inline tests/testthat/test-ArmadilloOAuth.R:14]: does not verify very much yet

File: R/ArmadilloOAuth.R [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+#'
+#' @param armadillo_server url of the Armadillo server
+#'
+#' @importFrom httr GET stop_for_status content
+#' @importFrom urltools path
+#'
+#' @return a dataframe with issuerUrl and clientId
+#'
+#' @noRd
+#' @keywords internal
```
[the remark this answers is not in the document]
Comment (sidohaakma) [reply R/ArmadilloOAuth.R:36]: done!

File: R/utils.R [hunk 5]
```diff
@@ -73,6 +122,11 @@
   data_frame
 }
 
+#' Deparse expression to check if it's R code/functions
```
[the remark this answers is not in the document]
Comment (sidohaakma) [reply R/utils.R:125]: updated the description

File: vignettes/DSArmadillo.Rmd.orig [hunk 6]
```diff
@@ -0,0 +1,316 @@
...
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+# DataSHIELD Armadillo
+When you start to use Aramdillo as a backend for DataSHIELD you can use the `DSMolgenisArmadillo` package for research purposes.
+
+## Installation
+The DataSHIELD Armadillo package is available on CRAN. You can install it by executing the following code-block:
```
[the remark this answers is not in the document]
Comment (sidohaakma) [reply vignettes/DSArmadillo.Rmd.orig:21]: updated docs

File: vignettes/DSArmadillo.Rmd.orig [hunk 7]
```diff
@@ -0,0 +1,316 @@
...
+
+```{r, eval = FALSE}
+install.packages("DSMolgenisArmadillo")
+```
+
+Make sure you install the DataSHIELD client (`dsBaseClient`) to perform the actual analysis.
```
[the remark this answers is not in the document]
Comment (sidohaakma) [reply vignettes/DSArmadillo.Rmd.orig:27]: updated docs and DESCRIPTION

File: vignettes/DSArmadillo.Rmd.orig [hunk 8]
```diff
@@ -0,0 +1,316 @@
...
+- Methods and packages
+
+All of these components will be addressed.
+
+### Login on DataSHIELD Armadillo
+The login procedure consists of building a login dataframe and performing the login on the Armadillo server. The important part is to specify the driver. This should be `ArmadilloDriver`. 
+
+We support authentication 2 flows. 
+- as normal user (using oauth on the central authentication server)
+- as superuser (using basic authentication, username and password based)
```
[the remark this answers is not in the document]
Comment (sidohaakma) [reply vignettes/DSArmadillo.Rmd.orig:75]: removed the basic auth

File: vignettes/DSArmadillo.Rmd.orig [hunk 9]
```diff
@@ -0,0 +1,316 @@
...
+               url = paste0(armadillo_url, "?workspace=gecko/patient"),
+               user = "",
+               password = "",
+               token = token,
+               table = "datashield.PATIENT",
+               driver = "ArmadilloDriver")
+```
+
+```{r}
+# method 2: using basic authentication
```
[the remark this answers is not in the document]
Comment (sidohaakma) [reply vignettes/DSArmadillo.Rmd.orig:100]: removed the basic auth

File: vignettes/DSArmadillo.Rmd.orig [hunk 10]
```diff
@@ -0,0 +1,316 @@
...
+You can assign values from tables to symbols, but hardcoded values as well.
+```{r}
+# assign random data to 'K'
+datashield.assign.expr(conns = conns, symbol = "K", "c(10,50,100)")
+# calculate the mean of 'K' to see that the assignment has worked
+ds.mean("K", datasources = conns)
+```
+
+#### Table methods
+When up assigned symbols in R you check which tables (`data.frame`'s) are available on the Armadillo.
```
[the remark this answers is not in the document]
Comment (sidohaakma) [reply vignettes/DSArmadillo.Rmd.orig:162]: fixed

**Pull request merged**